index.js

```diff
@@ -36,22 +36,38 @@ function normalizeName(problemName) {
   return problemName.toLowerCase().replace(/\s/g, '_');
 }
 
-async function commit(octokit, owner, repo, defaultBranch, commitInfo, treeSHA, latestCommitSHA, submission) {
+async function commit(params) {
+  const {
+    octokit,
+    owner,
+    repo,
+    defaultBranch,
+    commitInfo,
+    treeSHA,
+    latestCommitSHA,
+    submission,
+    destinationFolder
+  } = params;
+
   const name = normalizeName(submission.title);
   log(`Committing solution for ${name}...`);
 
   if (!LANG_TO_EXTENSION[submission.lang]) {
     throw `Language ${submission.lang} does not have a registered extension.`;
   }
 
+  const path = !!destinationFolder
```

Comment:
nit: I think it'd be slightly cleaner to only have to do the `problems/${name}/solution.${LANG_TO_EXTENSION[submission.lang]}` part once

One possible way would be:

```
const prefix = !!destinationFolder ? `{destinationFolder}/` : '';
const path = `{prefix}problems/${name}/solution.${LANG_TO_EXTENSION[submission.lang]}`
```